src/dsp/ethereum/mpt_data_prep.ts

```diff
@@ -0,0 +1,151 @@
+import { utils } from 'ethers'
+import type { BlockPrep } from './blockprep'
+
+function i32ToLittleEndianHexStr(value: number) {
+    const buffer = Buffer.alloc(8);
+    buffer.writeUInt32LE(value, 0);
+    return buffer.toString('hex');
+}
+
+function prependLengthInLittleEndianHex(hexString: string) {
+    const lengthInBytes = hexString.length / 2;
+    const lengthAsLittleEndianHex = i32ToLittleEndianHexStr(lengthInBytes);
+    return lengthAsLittleEndianHex + hexString;
+}
+
+function formatProofPath(rawProofPath: string) {
+    let proofPath = ''
+    if (rawProofPath.startsWith('0x')) {
+        proofPath = rawProofPath.slice(2);
+    } else {
+        proofPath = rawProofPath;
+    }
+    const newHexString = prependLengthInLittleEndianHex(proofPath);
+    return newHexString;
+}
+
+function safeHex(rawHex: string) {
+    let hex = ''
+    if (rawHex.startsWith('0x')) {
+        hex = rawHex.slice(2);
+    } else {
+        hex = rawHex;
+    }
+    if(hex.length % 2 == 0) {
+        return hex;
+    } else {
+        return '0' + hex;
+    }
+}
+
+function padHexStringToU64LengthLittleEndian(rawHex: string) {
+    const length = rawHex.length;
+    const remainder = length % 16;
+    if (remainder === 0) return rawHex;
+
+    const paddingSize = 16 - remainder;
+    const paddedHexString = rawHex + '0'.repeat(paddingSize);
+    return paddedHexString;
+}
+
+function hexToLittleEndian(hexString: string) {
+    let result = hexString.match(/../g).reverse().join('');
+    return result;
+}
+
+export class MptInputPrep {
```

Comment:
it doesn't look like a prep if there's only data processing and no data fetching

Reply:
oh i see, it's called InputPrep not data prep. ok make sense. just that the file name is misleading